lib/app/features/text_size/text_size_controller.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:black_tax_and_white_benefits/app/features/settings/shared_preferences.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'text_size_controller.g.dart';
+
+@Riverpod(keepAlive: true)
+class TextSizeController extends _$TextSizeController {
+  double build() {
+    final prefs = ref.watch(sharedPreferencesProvider);
+    return prefs.valueOrNull?.getDouble(textSizeKey) ?? 1.0;
+  }
+
+  static const String textSizeKey = 'textSize';
+
+  Future<void> _storeState() async {
+    final prefs = ref.read(sharedPreferencesProvider);
+    if (prefs.hasValue) {
```

Comment:
I think you are looking for requireValue

Reply:
Doesn't requireValue throw an error if the value is null? Do we want that?

Reply:
@alexdivadi  remember this one? Please make a jira task to use this one for firebase, database and shared prefs initialization...plz make a note if it "messes up" integration tests or widget tests you can change the spinner to "Loading" in tests, minor weirdness.
butdon't forgett, thi si sgreat, it looks awesome imho

Reply:
sorry, forgot tlink https://codewithandrea.com/articles/robust-app-initialization-riverpod/
anyway, requireValue is pretty good if you "know" you initialized it on app startup. https://riverpod.dev/docs/essentials/eager_initialization